### PR TITLE
[gdal] Fix usage and add dependencies on UNIX

### LINKED ISSUE #15526: [gdal:x64-windows-static] missing geos dependencies
**Host Environment**
- OS: Windows 10 build 1607
- Compiler: Visual C++ 2017 00370-20001-21499-AA108 / VisualStudio.15.Release/15.9.29+28307.1300

**To Reproduce**
- `.\vcpkg install gdal --triplet=x64-windows-static`
- Write a basic program using gdal and compile/link with cmake.

Linking fails with missing symbols errors from the geos library.
Manually adding the geos package with cmake solves the linking errors (well... almost as it may generate conflicts with windows's `io.h` and `./include/geos/io.h` but that's another story).

Looking at `./ports/gdal/dependency_win.cmake` we see these declarations for geos
```
  # Setup geos libraries + include path
  set(GEOS_INCLUDE_DIR "${CURRENT_INSTALLED_DIR}/include" )
  set(GEOS_LIBRARY_REL "${CURRENT_INSTALLED_DIR}/lib/geos_c.lib" )
  set(GEOS_LIBRARY_DBG "${CURRENT_INSTALLED_DIR}/debug/lib/geos_cd.lib" )
```

But when geos is installed, it comes with two static libs, not one:
2021-01-08 16_10_05-lib

So I guess the geos declarations should be modified this way to be complete:
```
  # Setup geos libraries + include path
  set(GEOS_INCLUDE_DIR "${CURRENT_INSTALLED_DIR}/include" )
  set(GEOS_LIBRARY_REL "${CURRENT_INSTALLED_DIR}/lib/geos_c.lib ${CURRENT_INSTALLED_DIR}/lib/geos.lib" )
  set(GEOS_LIBRARY_DBG "${CURRENT_INSTALLED_DIR}/debug/lib/geos_cd.lib ${CURRENT_INSTALLED_DIR}/debug/lib/geosd.lib" )
```

### PR DESCRIPTION
**Pull request by JackBoosY**
Since gdal relies on many ports, and using static gdal library under Unix also needs to use these dependencies, fix the dependencies in Unix.
Add required dependencies giflib and zstd.

Depends on #16085.
Related: #15375.

### CODE REVIEW

File: ports/gdal/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -8,5 +8,54 @@ find_library(GDAL_LIBRARY_RELEASE NAMES gdal_i gdal NAMES_PER_DIR PATH_SUFFIXES
 
 select_library_configurations(GDAL)
 
+if (UNIX)
```
Comment (ras0219) [inline ports/gdal/vcpkg-cmake-wrapper.cmake:11]: This should probably switch based on static/dynamic, not on `UNIX`. This likely means the file should be renamed to `vcpkg-cmake-wrapper.cmake.in`, this line should become `if("@VCPKG_LIBRARY_LINKAGE@" STREQUAL "static")`, and we should use `configure_file( @ONLY)`.

Comment (JackBoosY) [reply ports/gdal/vcpkg-cmake-wrapper.cmake:11]: @ras0219 I think this suitable for `WINDOWS & dynamic` and `LINUX & static`.
What do you think about?

**Comment by JackBoosY** [comment]
Waiting for merge #16138.

**Comment by ras0219-msft** [comment]
This may also be fixed by #15808

**Comment by JackBoosY** [comment]
@ras0219-msft Since your PR may last a long time, do I need to continue the PR?

**Comment by JackBoosY** [comment]
Replaced by #15808.

**Pull request closed without merging**